fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/HashJobStore.java

```diff
@@ -111,9 +111,20 @@ private boolean satisfiesFilter(JobRow jobRow, JobRowFilter jobRowFilter) throws
             );
         }
 
-        if (!jobRowFilter.getValues().contains(jobRow.get(filterJobField))) {
-            return false;
+        switch (filterOperation) {
+            case notin:
+                return !filterValues.contains(actualValue);
+            case startswith:
+                return filterValues.stream().anyMatch(filterValue -> actualValue.startsWith(filterValue));
+            case contains :
+                return filterValues.stream().anyMatch(filterValue -> actualValue.contains(filterValue));
+            case in:
```

Comment:
We should probably include a comment here indicating that the fall-through is intentional because `in` is a synonym for `eq`. I believe we do that in other places as well.